resources/startup.sh

```diff
@@ -38,61 +36,78 @@ if [[ ! -d ${OPENLDAP_CONFIG_DIR}/cn=config ]]; then
+  LDAP_ROOTPASS_ENC=$(slappasswd -s "$LDAP_ROOTPASS")
+  export LDAP_ROOTPASS_ENC
+
   LDAP_DOMAIN=$(doguctl config --global domain)
+  export LDAP_DOMAIN
 
   echo "get admin user details"
   CONFIG_USERNAME=$(doguctl config "admin_username")
   ADMIN_USERNAME=${CONFIG_USERNAME:-admin}
+  export ADMIN_USERNAME
 
   ADMIN_MAIL=$(doguctl config "admin_mail") ||  ADMIN_MAIL="${ADMIN_USERNAME}@${DOMAIN}"
+  export ADMIN_MAIL
 
   CONFIG_GIVENNAME=$(doguctl config "admin_givenname") || CONFIG_GIVENNAME="admin"
   ADMIN_GIVENNAME=${CONFIG_GIVENNAME:-CES}
+  export ADMIN_GIVENNAME
 
   CONFIG_SURNAME=$(doguctl config "admin_surname") || CONFIG_SURNAME="admin"
   ADMIN_SURNAME=${CONFIG_SURNAME:-Administrator}
+  export ADMIN_SURNAME
 
   CONFIG_DISPLAYNAME=$(doguctl config "admin_displayname") || CONFIG_DISPLAYNAME="admin"
   ADMIN_DISPLAYNAME=${CONFIG_DISPLAYNAME:-CES Administrator}
+  export ADMIN_DISPLAYNAME
 
   echo "get manager and admin group name"
   MANAGER_GROUP=$(doguctl config --global "manager_group") || MANAGER_GROUP="cesManager"
+  export MANAGER_GROUP
   ADMIN_GROUP=$(doguctl config --global admin_group) || ADMIN_GROUP="cesAdmin"
+  export ADMIN_GROUP
   ADMIN_MEMBER=$(doguctl config admin_member) || ADMIN_MEMBER="false"
 
   echo "get admin password"
   # TODO remove from etcd ???
   CONFIG_PASSWORD=$(doguctl config -e "admin_password")
   ADMIN_PASSWORD=${CONFIG_PASSWORD:-admin}
-  ADMIN_PASSWORD_ENC="$(slappasswd -s $ADMIN_PASSWORD)"
+  ADMIN_PASSWORD_ENC="$(slappasswd -s "${ADMIN_PASSWORD}")"
+  export ADMIN_PASSWORD_ENC
 
   mkdir -p ${OPENLDAP_CONFIG_DIR}
 
   if [[ ! -s ${OPENLDAP_ETC_DIR}/ldap.conf ]]; then
     echo "rendering ldap.conf template"
-    render_template /srv/openldap/conf.d/ldap.conf.tpl > ${OPENLDAP_ETC_DIR}/ldap.conf
+    doguctl template /srv/openldap/conf.d/ldap.conf.tpl ${OPENLDAP_ETC_DIR}/ldap.conf
   fi
 
   if [[ ! -s ${OPENLDAP_ETC_DIR}/slapd-config.ldif ]]; then
     echo "rendering slapd-config.ldif template"
-    render_template /srv/openldap/conf.d/slapd-config.ldif.tpl > ${OPENLDAP_ETC_DIR}/slapd-config.ldif
+    doguctl template /srv/openldap/conf.d/slapd-config.ldif.tpl ${OPENLDAP_ETC_DIR}/slapd-config.ldif
   fi
 
   slapadd -n0 -F ${OPENLDAP_CONFIG_DIR} -l ${OPENLDAP_ETC_DIR}/slapd-config.ldif > ${OPENLDAP_ETC_DIR}/slapd-config.ldif.log
 
   mkdir -p ${OPENLDAP_BACKEND_DIR}/run
   chown -R ldap:ldap ${OPENLDAP_BACKEND_DIR}
   chown -R ldap:ldap ${OPENLDAP_CONFIG_DIR} ${OPENLDAP_BACKEND_DIR}
+  mkdir -p ${OPENLDAP_RUN_DIR}
+  chown -R ldap:ldap ${OPENLDAP_RUN_DIR}
 
+   #shellcheck disable=SC2044
+   #fixed with filecheck below
   if [[ -d /srv/openldap/ldif.d ]]; then
-    for f in $(find /srv/openldap/ldif.d -type f -name "*.tpl"); do
-      render_template $f > $(echo $f | sed 's/\.tpl//g')
+    for file in $(find /srv/openldap/ldif.d -type f -name "*.tpl"); do
+     [[ -e "$file" ]] || continue
+     # search for .tpl and replace it with empty
+     echo "${file//\.tpl/}";
```

Comment:
Why are you printing out a filename? Where has the template rendering gone?

Reply:
IDK. I guess I accidentally replaced this line with a test command from the terminal. I've added some testcases in goss to get immediate feedback when there are issues with the template rendering. https://github.com/cloudogu/ldap/pull/7/commits/d3f86091cc1f389c08ae0472ca5b5a6c66509cf0